r/src/bpcells-cpp/matrixIterators/ImportMatrixAnnDataHDF5.cpp

```diff
@@ -557,33 +557,19 @@ std::string get10xMatrixType(std::string file, std::string group) {
         return "double";
     }
     throw std::runtime_error(
-        "get10xMatrixType: unrecognized type for group " + group + " in file " + file
+        "getAnnDataMatrixType: unrecognized type for group " + group + " in file " + file
     );
 }
 
 bool isRowOrientedAnnDataMatrix(std::string file, std::string group) {
-    assertAnnDataSparse(file, group);
-    H5ReaderBuilder rb(file, group, 1024, 1024);
-
     HighFive::SilenceHDF5 s;
-    HighFive::Group &g = rb.getGroup();
-    std::string encoding;
+    HighFive::File h5file = openH5ForReading(file);
 
-    if (g.hasAttribute("h5sparse_format")) {
-        // Support for legacy format
-        g.getAttribute("h5sparse_format").read(encoding);
-        encoding += "_matrix";
-    } else if (g.hasAttribute("encoding-type")) {
-        g.getAttribute("encoding-type").read(encoding);
-    } else {
-        throw std::runtime_error(
-            "h5ad could not be read - missing attribute 'encoding-type' on sparse matrix"
-        );
-    }
+    std::vector<uint32_t> dims;
+    bool row_major, is_sparse;
+    readAnnDataDims(h5file, group, dims, row_major, is_sparse);
```

Comment:
I like how this is now a much cleaner function.  However, I think that using a void typed function to get the row_major status makes it a little bit unclear on how you obtained row_major status.  In terms of readability, having to look into the function call on `readAnnDataDims` to see that your values changed as they're passed by reference is an extra step.  Maybe we can just make it return the bool, or the tuple of bools?